eagle-jpm/eagle-jpm-mr-running/src/main/java/org/apache/eagle/jpm/mr/running/parser/MRJobEntityCreationHandler.java

```diff
@@ -78,24 +78,36 @@ public boolean flush() {
             eagleServiceConfig.username,
             eagleServiceConfig.password);
         client.getJerseyClient().setReadTimeout(eagleServiceConfig.readTimeoutSeconds * 1000);
-        try {
-            LOG.info("start to flush mr job entities, size {}", entities.size());
-            client.create(entities);
-            LOG.info("finish flushing mr job entities, size {}", entities.size());
-            entities.clear();
-        } catch (Exception e) {
-            LOG.warn("exception found when flush entities, {}", e);
-            e.printStackTrace();
-            return false;
-        } finally {
-            client.getJerseyClient().destroy();
+
+        int count = 0;
+        boolean success = false;
+        while (count++ < MAX_RETRY_COUNT && !success) {
             try {
-                client.close();
+                LOG.info("start to flush mr job entities, size {}", entities.size());
+                client.create(entities);
+                LOG.info("finish flushing mr job entities, size {}", entities.size());
+                entities.clear();
+                success = true;
+                LOG.info("Success of flushing mr job entities");
             } catch (Exception e) {
-                // ignored exception
+                LOG.warn("exception found when flush entities, {}", e);
+            } finally {
+                client.getJerseyClient().destroy();
```

Comment:
I do not think this retry logic will work. If flush failed first time, client will be destroyed or closed!